app/assets/javascripts/discourse/app/components/sidebar/categories-form-modal.js

```diff
@@ -13,9 +13,27 @@ export default class extends Component {
     ...this.currentUser.sidebar_category_ids,
   ];
 
-  @tracked siteCategories = this.site.categoriesList.filter((category) => {
-    return !category.parent_category_id && !category.isUncategorizedCategory;
-  });
+  categoryGroupings = [];
+
+  constructor() {
+    super(...arguments);
+
+    const filteredCategories = this.site.sortedCategories.filter(
+      (category) => !category.isUncategorizedCategory
+    );
+
+    filteredCategories.reduce((categoryGrouping, category, index, arr) => {
```

Comment:
filter and reduce are creating 2 iterations through all items. Maybe, it would be better to combined them and use guard in reduce, something like:
```
if(!category.isUncategorizedCategory) { return categoryGrouping } 
```

Reply:
Good point :+1: I've made the changes.